components/scream/src/physics/p3/p3_functions_f90.cpp

```diff
@@ -220,11 +228,13 @@ namespace p3 {
 // helper functions
 namespace {
 
-template <size_t N>
+template <size_t N, size_t M>
 void gen_random_data(const std::array<std::pair<Real, Real>, N>& ranges,
-                     const std::array<Real**, N>& ptrs,
+                     const std::array<Real**, M>& ptrs,
                      Real* data, Int nk)
 {
+  static_assert(N <= M);
```

Comment:
This is syntactically wrong; static_assert requires a message as a second arg. I'm curious why the GPU build isn't seeing that. Is this used only in CPU builds?

Reply:
Apparently this construction is allowed in C++17:

https://en.cppreference.com/w/cpp/language/static_assert

Though I don't know that this explains why it compiled with our settings.

Reply:
Oh, nice! Can't wait until 2017.

Reply:
Looks like only bowman caught this. Fixing now.